spec/support/models/post.cr

```diff
@@ -8,8 +8,11 @@ class Post < BaseModel
     column title : String
     column published_at : Time?
     has_many comments : Comment
+    has_many nice_comments : Comment,
+      query: Comment::BaseQuery.new.body(Comment::NICE_COMMENT_BODY)
     has_many taggings : Tagging
     has_many tags : Tag, through: [:taggings, :tag]
+    has_many funny_tags : Tag, through: [:taggings, :tag], query: Tag::BaseQuery.new.name(Tag::FUNNY_TAG_NAME)
```

Comment:
Maybe we don't allow the `query` on `through` to start? We could always add that in later 🤔